src/test/java/org/carlspring/cloud/storage/s3fs/S3FileChannelTest.java

```diff
@@ -56,78 +81,310 @@ public void tearDown()
         s3fsProvider.close((S3FileSystem) fileSystem);
     }
 
+    @ParameterizedTest(name = "{index} ==> position={0}, bytesRead={1}")
+    @MethodSource(value = "positionAndBytesReadForReadProvider")
```

Comment:
Does this make sense to be parameterized, if it's just for one value (or... am I misreading something)?

Reply:
Yes, it does make sense because that's not just a value, that's the name of the method `positionAndBytesReadForReadProvider`, which provides multiple arguments for the test. Take it into account that this is using `@MethodSource` annotation instead of `@ValueSource` annotation.

Reply:
Ah, right! I see, thanks for the clarification! :)